### PR TITLE
Documentation for additional variables attributes.

### PR DESCRIPTION
**Pull request by fviale**
Added groups, description, advanced, and hidden attributes definition

Added SpEL functions associated with hidden variables.

Added a dynamic form workflow example.

### CODE REVIEW

**Review by aminelouati — APPROVED** [review]

**Review by mklkun — APPROVED** [review]

File: src/docs/user/references/Variables.adoc [hunk 1]
```diff
@@ -16,14 +16,21 @@ only within the task execution context.
...
-
+* *name*: the name of the variable.
+* *value*: the value of the variable.
+* *model*: the type of the value (optional). See section <<_variable_model,Variable Model>>.
+* *description*: a description of the variable (optional). The description can be written using html syntax (e.g. `This variable is <b>mandatory</b>.`).
+* *group*: a group name can be associated with the variable (optional). When the workflow has many variables,
+defining variable groups helps understanding the various parameters.
+* *advanced*: when a variable is advanced, it is not displayed by default in the workflow submission form.
+A checkbox allows the user to display or hide advanced variables.
+* *hidden*: a hidden variable is not displayed to the user in the workflow submission form. Hidden variables have the following uses:
```
Comment (mboussaa) [inline src/docs/user/references/Variables.adoc:31]: usages

File: src/docs/user/references/Variables.adoc [hunk 2]
```diff
@@ -16,14 +16,21 @@ only within the task execution context.
...
-In a workflow you can define *job variables* that are shared and visible by all tasks.
+In a workflow, you can define *job variables* that are shared and visible by all tasks.
 
 A job variable is defined using the following attributes:
 
-* *name*: the name of the variable
-* *value*: the value of the variable
-* *model*: the type of the value (optional). See section <<_variable_model,Variable Model>>
-
+* *name*: the name of the variable.
```
Comment (mboussaa) [inline src/docs/user/references/Variables.adoc:23]: the variable name,
the variable value 
the variable type or model
the variable description
the group name

**Comment by mboussaa** [comment]
could u also say that default variables are now called main variables?

File: src/docs/user/references/Variables.adoc [hunk 3]
```diff
@@ -226,12 +242,45 @@ Note that #value always contains a string and must be converted to other types i
...
+Consider the following variables definition:
+
+```xml
+  <variables>
+    <variable name="type" value="vegetable" model="PA:LIST(vegetable,fruit)" description="" group="" advanced="false" hidden="false"/>
+    <variable name="potatoes" value="0" model="PA:INTEGER" description="Amount of potatoes to order (in kilograms)" group="vegetables" advanced="false" hidden="false"/>
+    <variable name="leek" value="0" model="PA:INTEGER" description="Amount of leek to order (in kilograms)" group="vegetables" advanced="false" hidden="false"/>
+    <variable name="apples" value="0" model="PA:INTEGER" description="Amount of apples to order (in kilograms)" group="fruits" advanced="false" hidden="true"/>
+    <variable name="oranges" value="0" model="PA:INTEGER" description="Amount of oranges to order (in kilograms)" group="fruits" advanced="false" hidden="true"/>
+    <variable name="type_handler" value="" model="PA:SPEL(variables[&#39;type&#39;] == &#39;vegetable&#39; ? showGroup(&#39;vegetables&#39;) &amp;&amp; hideGroup(&#39;fruits&#39;) : showGroup(&#39;fruits&#39;) &amp;&amp; hideGroup(&#39;vegetables&#39;))" description="" group="" advanced="false" hidden="true"/>
```
Comment (tao-xinxiu) [inline src/docs/user/references/Variables.adoc:258]: Replace `&#39;` back to quote could be more clear for the document.

**Review by tao-xinxiu — APPROVED** [review]

File: src/docs/user/references/Variables.adoc (hunk 3 above)
Comment (fviale) [reply src/docs/user/references/Variables.adoc:258]: ok

**Pull request merged**